src/dmd/expressionsem.d

```diff
@@ -8662,7 +8662,9 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         else if (auto se = exp.e1.isSliceExp())
         {
             Type tn = se.type.nextOf();
-            if (exp.op == TOK.assign && !tn.isMutable())
+            const fun = sc.func;
+            if (exp.op == TOK.assign && !tn.isMutable() &&
```

Comment:
I suggest adding a link to the bugzilla issue in the form of a comment next to or above this code.  It helps to explain to the next person why it's there.

Reply:
Or, perhaps a comment describing the unique privileges of static constructors.

Reply:
Why was this marked as resolved?

Reply:
Because two lines down.